src/state.rs

```diff
@@ -151,18 +153,23 @@ pub struct VMState {
     pub events: Vec<Event>,
     pub register_context_u128: u128,
     pub default_aa_code_hash: [u8; 32],
+    pub hook_address: u32,
+    pub use_hooks: bool,
 }
 
 // Totally arbitrary, probably we will have to change it later.
 pub const DEFAULT_INITIAL_GAS: u32 = 1 << 16;
 impl VMState {
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
         program_code: Vec<U256>,
         calldata: Vec<u8>,
         contract_address: H160,
         caller: H160,
         context_u128: u128,
         default_aa_code_hash: [u8; 32],
+        hook_address: u32,
+        use_hooks: bool,
```

Comment:
Maybe these could be an option?
```suggestion
        hook_address: Option<u32>,
```
`None` would imply you don't use hooks.

Reply:
I feel that would make the code uglier when checking if it needs to use hooks
Instead of doing `if use_hooks ...` it would have to do `if hook_address.is_some()` or something like
`if let Some(address) = hook_address ...`

Reply:
It's uglier but it's less error prone, since you only have access to the address if you already checked you're using hooks.